src/kudu/security/tls_context.cc

```diff
@@ -594,8 +600,65 @@ Status TlsContext::LoadCertFiles(const string& ca_path,
   return Status::OK();
 }
 
-Status TlsContext::InitiateHandshake(TlsHandshakeType handshake_type,
-                                     TlsHandshake* handshake) const {
+// Encode the list of alpn protocols into wire format
+static std::vector<unsigned char> encodeAlpn(const std::vector<std::string>& alpns) {
+  std::vector<unsigned char> result;
+  for (auto p : alpns) {
+    CHECK(p.length() <= 255);
+    result.push_back((unsigned char)(p.length()));
+    for (auto c : p) {
+      result.push_back((unsigned char)c);
+    }
+  }
+  return result;
+}
+
+Status TlsContext::SetSupportedAlpns(
```

Comment:
How does sending Alpn as a part of the handshake help us to get rid of
erstwhile protobuf based kuduraft negotiation?

Reply:
See comments in tls_context.h.